### PR TITLE
Fixes VSTS Bug 951152: [Feedback] Visual Studio Commmunity lacks UTF-8

### PR DESCRIPTION
**Pull request by mkrueger**
(no signature) format

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/951152

Needed to switch away from code page representation. There is no code
page for an UTF-8 encoding without BOM.

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Dialogs/FileSelectorDialog.cs [hunk 1]
```diff
@@ -107,18 +107,18 @@ public FileSelectorDialog (string title, Gtk.FileChooserAction action): base (ti
 			if (action != Gtk.FileChooserAction.Open)
 				closeWorkspaceCheck.Visible = ShowViewerSelector = false;
 		}
-		
-		public int SelectedEncoding {
+
+		public TextEncoding SelectedEncoding {
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Dialogs/FileSelectorDialog.cs]: This API change will break windows builds.

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Dialogs/FileSelectorDialog.cs]: This class is not even public...

**Review by Therzok — APPROVED** [review]
Seems like windows is not affected. My initial assumption was wrong, sorry!

**Review by sevoku — APPROVED** [review]

**Comment by sevoku** [comment]
@monojenkins backport release-8.3

**Pull request merged**